### DISCUSSION TITLE
error while deploing next js app .

### DISCUSSION THREAD

**Discussion by shahebazshaikh99** [vercel/next.js#55302] [Help]
### Summary

in my react js app when i have to deploy it on server normally i run the command 'npm run build' and build folder ipload it on server and thats working fine on server but when i hdeploy my next js getServerSideProps app i run command 'npm run build' and upload my .next file on server it shows 'error' but while i upload my entire project directory like src, public, .next and etc. that;s working fine on server but size of the files is 700mb which is latger.
what is the problem that my .next file is not working on server?
and what is the standard way ?

**Comment by icyJoseph** [comment]
- Are you using `export` mode? 
- When you have `getServerSideProps` you have to include a bunch of other dependencies to the project. People often use the standalone mode instead. https://nextjs.org/docs/app/api-reference/next-config-js/output#automatically-copying-traced-files

**Reply by shahebazshaikh99** [reply]
iam using command npm run build production and then entire directory with include src, public , and etc upload on aws server then its working fine.
but when i upload only .next file then install node modules on server then it shows error??
so its a  statndard way to upload all directory to to server?

**Reply by icyJoseph** [reply]
Do you copy the package.json? 

I think it is best to see how the Docker example does it, regardless of docker usage or not: https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile

You see that they use a builder to generate the `.next` folder, with `standalone` mode activated, and then copy public, `.next/static`, and `/app/.next/standalone`.

> Additionally, a minimal server.js file is also output which can be used instead of next start. This minimal server does not copy the public or .next/static folders by default as these should ideally be handled by a CDN instead, although these folders can be copied to the standalone/public and standalone/.next/static folders manually, after which server.js file will serve these automatically.

I don't think you need the entire thing though, but it is usually not so straightforward to figure out what to include. That's why the `standalone` mode exists.
